### ISSUE TITLE
Update Sphinx build process to be versioned

### ISSUE THREAD

**Issue by iancze** [MPoL-dev/MPoL#38]
**Is your feature request related to a problem or opportunity? Please describe.**
Maintaining only a single "main" branch version of the docs has its downsides as we grow the MPoL userbase. Mainstream users will want to install (and keep using) a stable `pip` API, developers will want to reference documentation to the latest (tagged?) main branch. Forcing everyone to use the same, most recent version of the docs is bound to lead to frustration.

The problem is that this versioning functionality is not provided natively by Sphinx, the docs build system. Instead, it is provided by wrapper scripts from readthedocs. So, while our current docs build on github actions has made the build a lot easier now that we've left readthedocs, we do miss this versioning functionality.

**Describe the solution you'd like**
Investigate Sphinx extensions/scripts designed to work outside of readthedocs. 
[sphinx-multiversion](https://holzhaus.github.io/sphinx-multiversion/master/context.html) looks like the most promising solution, [doctr-versions-menu](https://github.com/goerz/docs_versions_menu) might be another solution. And possibly [unladen](https://github.com/dfm/unladen).

Create separate build branches for tagged releases (i.e., those registered with PyPI) and the main branch.

**Comment by iancze** [comment]
Closing this as low-priority for the foreseeable future. v1.0 (and the assumption of a stable API) is sufficiently far off that MPoL and its API are considered at-risk as new features are developed and stabilizied. For more info, see [Semantic Versioning](https://semver.org/). 

Beta users are encouraged to upgrade as new versions are pushed to main, since these usually include bug fixes.

**Comment by iancze** [comment]
Ok, I take that back. I think there is a need to provide users with a stable tagged version of the package and documentation for it that people can build off of for a specific project while also allowing development on code + docs to proceed in parallel. 

It would be nice to have this type of tagging functionality through Sphinx, but a quick look around at options and the only package that I could find that is recently maintained is [sphinx-versioned-docs](https://sphinx-versioned-docs.readthedocs.io/en/latest/index.html) and it doesn't list our theme as compatible (yet).

I think the best option now is to update the docs to include
* how to install the latest *tagged* version of a release and pin to a specific version
* how to build the docs locally for that, if you need to access them
* recommendation that the user review recent updates on `main` and corresponding entries in changelog.md as they upgrade as the package develops.
